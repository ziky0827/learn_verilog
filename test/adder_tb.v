//! Code comment
`include "adder.v"    //包含的子模块
`timescale 1ns/100ps

module adder_tb();
    reg [0:1] A , B;
    reg Cin;
    output [0:1]S;
    output Cout;

    adder U(
              .X(A),  //!duankou
              .Y(B),
              .Cin(Cin),
              .Cout(Cout),
              .S(S)
          );

    initial
    begin
        A = 0;
        B = 2;
        Cin = 1;
        #10 A = 1;
        #10 Cin = 1;
        #10 B = 0;
        #20 A = 3;
        #30 Cin = 0;
        #40 B = 3;
        #50 Cin = 1;
    end

    initial
    begin
        $monitor($time,,,"part:%b  %b", S, Cout);
        #70
        $finish;
    end
    /*iverilog */
    initial
    begin
        $dumpfile("wave.vcd");        //生成的vcd文件名称
        $dumpvars(0);    //tb模块名称
    end
    /*iverilog */

endmodule
//!adder.v:44: $finish called at 70 (1s)不知道这个输出是什么意思
