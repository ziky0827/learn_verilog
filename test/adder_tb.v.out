#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 10;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_0000028203546580 .scope module, "adder_tb" "adder_tb" 2 5;
 .timescale -9 -10;
v0000028203544270_0 .var "A", 0 1;
v0000028203544310_0 .var "B", 0 1;
v00000282035443b0_0 .var "Cin", 0 0;
v0000028203544450_0 .net "Cout", 0 0, v00000282035468a0_0;  1 drivers
v00000282035444f0_0 .net "S", 0 1, v0000028203546940_0;  1 drivers
S_0000028203546710 .scope module, "U" "adder" 2 11, 3 4 0, S_0000028203546580;
 .timescale 0 0;
    .port_info 0 /INPUT 2 "X";
    .port_info 1 /INPUT 2 "Y";
    .port_info 2 /INPUT 1 "Cin";
    .port_info 3 /OUTPUT 1 "Cout";
    .port_info 4 /OUTPUT 2 "S";
v0000028203536580_0 .net "Cin", 0 0, v00000282035443b0_0;  1 drivers
v00000282035468a0_0 .var "Cout", 0 0;
v0000028203546940_0 .var "S", 0 1;
v0000028203544130_0 .net "X", 0 1, v0000028203544270_0;  1 drivers
v00000282035441d0_0 .net "Y", 0 1, v0000028203544310_0;  1 drivers
E_00000282034f9c60 .event anyedge, v0000028203536580_0, v00000282035441d0_0, v0000028203544130_0;
    .scope S_0000028203546710;
T_0 ;
    %wait E_00000282034f9c60;
    %load/vec4 v0000028203544130_0;
    %pad/u 3;
    %load/vec4 v00000282035441d0_0;
    %pad/u 3;
    %add;
    %load/vec4 v0000028203536580_0;
    %pad/u 3;
    %add;
    %split/vec4 2;
    %store/vec4 v0000028203546940_0, 0, 2;
    %store/vec4 v00000282035468a0_0, 0, 1;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_0000028203546580;
T_1 ;
    %pushi/vec4 0, 0, 2;
    %store/vec4 v0000028203544270_0, 0, 2;
    %pushi/vec4 2, 0, 2;
    %store/vec4 v0000028203544310_0, 0, 2;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000282035443b0_0, 0, 1;
    %delay 100, 0;
    %pushi/vec4 1, 0, 2;
    %store/vec4 v0000028203544270_0, 0, 2;
    %delay 100, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000282035443b0_0, 0, 1;
    %delay 100, 0;
    %pushi/vec4 0, 0, 2;
    %store/vec4 v0000028203544310_0, 0, 2;
    %delay 200, 0;
    %pushi/vec4 3, 0, 2;
    %store/vec4 v0000028203544270_0, 0, 2;
    %delay 300, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000282035443b0_0, 0, 1;
    %delay 400, 0;
    %pushi/vec4 3, 0, 2;
    %store/vec4 v0000028203544310_0, 0, 2;
    %delay 500, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000282035443b0_0, 0, 1;
    %end;
    .thread T_1;
    .scope S_0000028203546580;
T_2 ;
    %vpi_call 2 35 "$monitor", $time, " ", " ", "part:%b  %b", v00000282035444f0_0, v0000028203544450_0 {0 0 0};
    %delay 700, 0;
    %vpi_call 2 37 "$finish" {0 0 0};
    %end;
    .thread T_2;
    .scope S_0000028203546580;
T_3 ;
    %vpi_call 2 42 "$dumpfile", "wave.vcd" {0 0 0};
    %vpi_call 2 43 "$dumpvars", 32'sb00000000000000000000000000000000 {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "adder_tb.v";
    "./adder.v";
